cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -108,15 +111,19 @@ protected AbstractContext(Program program, RunId runId, Arguments arguments,
     this.pluginInstantiator = pluginInstantiator;
   }
 
-  public List<Id> getOwners() {
+  private List<Id> createOwners(Id.Program programId, @Nullable AdapterDefinition adapterSpec) {
     ImmutableList.Builder<Id> result = ImmutableList.builder();
-    result.add(program.getId());
+    result.add(programId);
```

Comment:
It's the existing logic (I just change it to use arguments other than fields). I think the adapter name is added below.